lib/temporal/connection/grpc.rb

```diff
@@ -633,11 +634,43 @@ def get_system_info
       attr_reader :url, :identity, :credentials, :options, :poll_mutex, :poll_request
 
       def client
-        @client ||= Temporalio::Api::WorkflowService::V1::WorkflowService::Stub.new(
+        return @client if @client
+
+        channel_args = {}
+
+        if options[:keepalive_time_ms]
+          channel_args["grpc.keepalive_time_ms"] = options[:keepalive_time_ms]
+        end
+
+        if options[:retry_connection]
+          channel_args["grpc.enable_retries"] = 1
+
+          channel_args["grpc.service_config"] = ::JSON.generate(
+            methodConfig: [
+              {
+                name: [
+                  {
+                    service: "temporal.api.workflowservice.v1.WorkflowService",
+                  }
+                ],
+                retryPolicy: {
+                  retryableStatusCodes: ["UNAVAILABLE"],
+                  maxAttempts: 3,
+                  initialBackoff: "0.1s",
+                  backoffMultiplier: 2.0,
+                  maxBackoff: "0.3s"
```

Comment:
@hughevans thanks for the PR! It looks good overall.

For maxAttempts, initialBackoff, backoffMultiplier, and maxBackoff - I think these are sane defaults, but if we're going to support retries I'd prefer that users be able to pass in their own values to override these if they want. Could you make them configurable?

Reply:
@DeRauk Thank you for checking this out. I’ve gone ahead and added a `retry_policy` option there too.

Also I looked at the README, but was a bit unsure where exactly documentation for this should go logically. Is that something you could take on after merging possibly? 🙇

Reply:
Thanks, and sure.